src/shared/install.c

```diff
@@ -889,8 +892,12 @@ static int find_symlinks_in_directory(
                                 free_and_replace(dest, x);
                         }
 
+                        q = path_extract_filename(dest, &fname);
+                        if (q < 0)
+                                return q;
```

Comment:
When `path_equal_filename()` is used, `path_extract_filename()` here is not necessary. Please drop this.